api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java

```diff
@@ -0,0 +1,252 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.acl.region;
+
+import com.epam.pipeline.controller.vo.region.AWSRegionDTO;
+import com.epam.pipeline.controller.vo.region.AbstractCloudRegionDTO;
+import com.epam.pipeline.entity.info.CloudRegionInfo;
+import com.epam.pipeline.entity.region.AbstractCloudRegion;
+import com.epam.pipeline.entity.region.AwsRegion;
+import com.epam.pipeline.entity.region.CloudProvider;
+import com.epam.pipeline.manager.region.CloudRegionManager;
+import com.epam.pipeline.security.acl.AclPermission;
+import com.epam.pipeline.test.acl.AbstractAclTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+public class CloudRegionApiServiceTest extends AbstractAclTest {
+
+    @Autowired
+    private CloudRegionApiService cloudRegionApiService;
+
+    @Autowired
+    private CloudRegionManager mockCloudRegionManager;
+
+    private final AbstractCloudRegionDTO cloudRegionDTO = new AWSRegionDTO();
+
+    private final List<String> availableCloudsList = Arrays.asList("AWS", "AZURE", "GCP");
+
+    private AwsRegion region;
+
+    private List<AbstractCloudRegion> singleRegionList;
+
+    private List<CloudRegionInfo> cloudRegionInfoList;
+
+    @Before
+    public void setUp() throws Exception {
+        region = new AwsRegion();
+        region.setId(1L);
+        region.setName("OWNER");
+        region.setOwner(OWNER_USER);
+
+        singleRegionList = new ArrayList<>();
+        singleRegionList.add(region);
+
+        cloudRegionInfoList = Collections.singletonList(new CloudRegionInfo(region));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldAllowLoadRegionsInfoForAdmin() {
+        doReturn(cloudRegionInfoList).when(mockCloudRegionManager).loadAllRegionsInfo();
+
+        assertThat(cloudRegionApiService.loadAllRegionsInfo()).isEqualTo(cloudRegionInfoList);
+    }
+
+    @Test
+    @WithMockUser(roles = GENERAL_USER_ROLE)
```

Comment:
Can't we use here `SIMPLE_USER_ROLE`? I think it is better to stuck to one role if both have equal meaning

Reply:
The method that is tested here has the condition ADMIN_OR_GENERAL_USER in PreAuthorize. I thought the General User had access as opposed to the Simple User. So I made a separate variable for General user.

Reply:
`General` user is any user with role named `USER` or `ROLE_USER`, so there is no actual difference between General user and Simple user